src/main/java/uk/gov/hmcts/reform/userprofileapi/service/impl/IdamServiceImpl.java

```diff
@@ -43,6 +45,11 @@ public IdamRegistrationInfo registerUser(IdamRegisterUserRequest requestData) {
             ResponseEntity<Object> entity = JsonFeignResponseHelper.toResponseEntity(response,
                     getResponseMapperClass(response, null));
             result = new IdamRegistrationInfo(entity);
+            HttpStatus httpStatus = HttpStatus.valueOf(response.status());
+            if (httpStatus.is5xxServerError()) {
+                result.setIdamRegistrationResponse(HttpStatus.UNAUTHORIZED);
+                result.setStatusMessage(IdamStatusResolver.MISSING_TOKEN);
```

Comment:
All 5xx cannot be due to missing token. This message could mislead the users. Can you please send appropriate message?